qwerty_mlir/lib/Qwerty/IR/QwertyOps.cpp

```diff
@@ -322,8 +327,24 @@ struct CallIndirectIf : public mlir::OpRewritePattern<qwerty::CallIndirectOp> {
         rewriter.eraseOp(else_yield);
 
         rewriter.replaceOp(call, new_if_op.getResults());
-        // Now that the old scf.if has no uses, we can finally erase it
-        rewriter.eraseOp(if_op);
+        if (if_op.getResults()[0].use_empty()) {
+            rewriter.eraseOp(if_op);
+        } else {
+            mlir::Operation *tgt = nullptr;
+            for (mlir::Operation *user : if_op->getUsers()) {
+                mlir::Operation *op = user;
+                while (op->getBlock() != if_op->getBlock()) {
+                    op = op->getParentOp();
```

Comment:
I would add an `assert(op && "Cannot move scf.if because downstream op has no parent in this block");`. It looks like `getParentOp()` will return `nullptr` if the op is top-level (like a `ModuleOp`): https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html#a8d9761f2b79a46d10ffb6ba28880cc2f. I think an assertion failure would be more helpful than a segfault